server/views/pages/approve/keyDates.njk

```diff
@@ -2,22 +2,37 @@
 
 {% macro keyDates(licence, isDueForEarlyRelease) %}
 
-    {% set releaseDate = licence.actualReleaseDate | datetimeToDateShort or licence.conditionalReleaseDate | datetimeToDateShort %}
+    {% set releaseDate = '<span class="urgent-highlight">' + licence.homeDetentionCurfewActualDate | datetimeToDateShort + '</span><span class="urgent-highlight urgent-highlight-message">HDC release</span>' if licence.kind === 'HDC' else licence.actualReleaseDate | datetimeToDateShort or licence.conditionalReleaseDate | datetimeToDateShort %}
 
     {% if isDueForEarlyRelease %}
         {% set releaseDateHtml = '<span class="urgent-highlight">' + releaseDate + '</span><span class="urgent-highlight urgent-highlight-message">Early release</span>' %}
     {% else %}
         {% set releaseDateHtml = '<p>' + releaseDate + '</p>' %}
     {% endif %}
 
+    {% set releaseDateHeading = "Release date/HDC start date" if licence.kind === 'HDC' else "Release date" %}
+
     {% set dateRows = [
             {
-                key: { text: "Release date" },
+                key: { text: releaseDateHeading, classes: "release-date-heading" },
                 value: { html: releaseDateHtml }
             }
         ]
      %}
 
+    {% if licence.kind === 'HDC' %}
+        {% set dateRows = (dateRows.push(
+                 {
+                    key: { text: "HDC end date" },
+                    value: { text: licence.conditionalReleaseDate | datetimeToDateShortMinusDay, classes: "hdc-end-date" }
```

Comment:
infact this should probably be calculated on the backend and part of the HDC licence response

Reply:
Should I implement that as part of this piece of work? And then remove the filter once it's part of the HDC licence response?